File: sdi3-180.Cli-SOAP/src/com/sdi/client/accion/DeshabilitarUsuario.java
package com.sdi.client.accion;

import com.sdi.ws.AdminService;
import com.sdi.ws.EjbAdminServiceService;
import com.sdi.ws.User;
import com.sdi.ws.UserStatus;

public class DeshabilitarUsuario{

	private AdminService adminService = new EjbAdminServiceService().getAdminServicePort();
	private User user = null;
	
	
	public void execute(Long id) {
		comprobarId(id);
		if (user == null){
			System.err.println("Usuario inexistente, no puede deshabilitarle");
			return;
		}
		if (user.getStatus() == UserStatus.CANCELLED){ 
			System.err.println("El usuario ya ha sido deshabilitado");
			return;
		}
		adminService.deshabilitarUsuario(id);
		System.out.println("Usuario deshabilitado corectamente");
	}
	
	public void comprobarId(Long id){
		for(User u : adminService.getUsuarios()){
			if(id.equals(u.getId()))
				user=u;
		}
	}

}

File: sdi3-180.Cli-SOAP-CSharp/sdi3-180.Cli-SOAP-CSharp/DeshabilitarUsuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_180.Cli_SOAP_CSharp
{
    class DeshabilitarUsuario
    {
        private EjbAdminServiceService adminService = new EjbAdminServiceService();
        private user user = null;


        public void execute(long id) {
		comprobarId(id);
		if (user == null){
			Console.WriteLine("Usuario inexistente, no puede deshabilitarle");
			return;
		}
		if (user.status == userStatus.CANCELLED){ 
			Console.WriteLine("El usuario ya ha sido deshabilitado");
			return;
		}
		adminService.deshabilitarUsuario(id, true);
		Console.WriteLine("Usuario deshabilitado corectamente");
	}

        public void comprobarId(long id){
            foreach (user u in adminService.getUsuarios()){
			    if(id.Equals(u.id))
				    user=u;
		    }
	    }
    }
}
